src/Pingalot/Layouts/ModernPing.cs

```diff
@@ -13,6 +13,13 @@ public static async Task<PingSession> StartAsync(PingRequestOptions options)
 			var pingRequestAgent = new PingRequestAgent();
 			var cancellationTokenSource = new CancellationTokenSource();
 
+			if (options.ExportFileFullPath != null)
+			{
+				var pingRequestFileExporter = new PingRequestFileExporter(options.ExportFileFullPath);
+				pingRequestAgent.PingCompleted += pingRequestFileExporter.ExportSingleResultToFile;
+
+			}
+
```

Comment:
What I mean by the last part is that some of the difficulty is really just how the classes call each other. Ideal world would have something like each "layout" just register events rather than handle the life cycle itself. The ping session is kinda redundant now as we don't do anything with it. Remove the replication from the cancel key press etc logic in each layout.

If I were to name what I'm thinking, something like "pipelines". Each thing (a layout, exporting, etc) is just an item on the pipeline. I guess kinda like how the middleware works for ASP.NET Core but more around events than just calling a `next();` action.